projects/ddata-ui-input/src/lib/components/select/select.component.spec.ts

```diff
@@ -1,426 +1,110 @@
-  });
 
-  it('setListModel() method should call hideListModal() method', () => {
-    const fakeSpy2 = spyOn(component, 'hideListModal');
-    component.setListModel();
-    expect(fakeSpy2).toHaveBeenCalled();
-  });
-
-  it('renderListComponent() method should do things', () => {
-    expect(component.renderListComponent()).toEqual(undefined);
-
-    const fakeSpy = spyOn(component, 'hideListModal');
-    const fakeSpy2 = spyOn(component.selectModel, 'emit');
-    const fakeSpy3 = spyOn((component as any), 'setSelected');
-    const fakeSpy4 = spyOn(component.selected, 'emit');
-
-    const fakeModel = new Tag().init();
-    fakeModel.id = 3 as ID;
-    component._selectedModel = fakeModel;
-    component._model = fakeModel;
-    (component as any).componentFactoryResolver = new FakeComponentFactoryResolver();
-    const fakeParameterDialog = {} as unknown as Type<any>;
-    component.dialogListContent = new DialogContentItem(fakeParameterDialog, component._selectedModel);
-    component.dialogSettings = new FakeDialogSettings();
-    component.dialogListHost = new FakeViewContainerRef();
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
 
-    component.renderListComponent();
-    expect(component.componentRef.instance.isModal).toBe(true);
-    expect(component.componentRef.instance.multipleSelectEnabled).toBe(component.dialogListContent.data.multipleSelectEnabled);
-    expect(component.componentRef.instance.isSelectionList).toBe(component.dialogListContent.data.isSelectionList);
-    expect(component.componentRef.instance.loadData).toBe(component.dialogListContent.data.loadData);
-    expect(component.componentRef.instance.filter).toEqual({});
-    expect(component.componentRef.instance.models).toBe(component.dialogListContent.data.models);
-    expect(component.componentRef.instance.datasArrived).toBe(undefined);
-    expect(component.componentRef.instance.selectedElements).toEqual([3]);
+    it('should select twice items', () => {
+        component.multipleSelect = true;
+        component._model = new MockModel();
+        component._field = 'items';
+        component.dialogSettings = {
+            createEditComponent: null,
+            listComponent: MockListComponent,
+            listOptions: {
+                models: [],
+                isModal: true,
+                multipleSelectEnabled: true,
+                isSelectionList: true,
+                loadData: false,
+                selectedElements: []
+            }
+        };
 
-    expect(fakeSpy).toHaveBeenCalled();
-    expect(fakeSpy2).toHaveBeenCalledWith('An array for this function');
-    expect(fakeSpy3).toHaveBeenCalled();
-    expect(fakeSpy4).toHaveBeenCalled();
+        component.showModal('list');
 
-    (component._model as any).id = [3 as ID, 5 as ID];
-    component.multipleSelect = true;
+        component.componentRef.instance.select.next([new MockModel().init({ name: 'test1' })]);
```

Comment:
a folyamat nem igazán jól van tesztelve. Jelenleg ezt teszteled:
1. létrehozod a komponenst
2. beállítod rajta a dialogSettings mezőt
3. kinyitod a dialogot
4. a komponenshez hozzáadsz egy kijelölt elemet

Helyette ezt kellene tesztelni:
1. létrehozod a komponenst
2. beállítod rajta a dialogSettings mezőt
2.1. beállítod a dialog beállításokban, hogy melyik model-ek vannak kijelölve
3. kinyitod a dialogot (elemre kattintással, nem property állítgatással)
4. ellenőrzöd, hogy ki van-e jelölve, aminek ki kell
5. bezárod a dialogot (elemre kattintással, nem property állítgatással)
6. ellenőrzöd, hogy a megfelelő helyen a HTML-ben megjelenik-e a kiválaszott adat
7. újra kinyitod (elemre kattintással)
8. próbálod kattintással módosítani a kiválasztottakat
9. bezárod
10. ellenőrzöd, hogy a megfelelő helyen a HTML-ben megjelenik-e a kiválaszott adat